packages/ui/src/composables/useStateful.ts

```diff
@@ -29,6 +29,19 @@ export function useStateful<T, D extends T = T> (
   defaultValue?: D,
 ) {
   const valueState = ref(defaultValue === undefined ? props.modelValue : defaultValue) as Ref<T>
+  const modelValueRef = toRef(props, 'modelValue')
+  const statefulRef = toRef(props, 'stateful')
+  let unwatchModelValue: Function
+
+  const watchModelValue = () => {
+    unwatchModelValue = watch(modelValueRef, (modelValue) => {
+      valueState.value = modelValue
+    })
+  }
+
+  watch(statefulRef, (stateful: boolean) => {
+    stateful ? watchModelValue() : unwatchModelValue?.()
+  }, { immediate: true })
```

Comment:
Maybe it will work the same?
```suggestion
  watch([props.stateful, props.modelValue], ([stateful: boolean, modelValue: T]) => {
    if (stateful && modelValue !== undefined) {
      valueState.value = modelValue
    }
  }, { immediate: true })
```

Reply:
This will never work because `props.stateful` and `props.modelValue` are not `refs`.

If we will make them `refs` to make it work there will be other issue - we will always watch for `modelValue` even when it is not needed.

Which is not optimal.

Reply:
This will never work because props.stateful and props.modelValue are not refs.

If we will make them refs, to make it work, there will be other issue - we will always watch for modelValue even when it is not needed.

Which is not optimal.

Reply:
I updated my suggestion: replaced props with functions:
`watch([() => props.stateful, () => props.modelValue], ([stateful, modelValue]) => {...`

Reply:
Ok then

"there will be other issue - we will always have additional watcher for modelValue in every component and change valueState twice when it is not needed.

Which is not optimal."

Reply:
Ok then

"there will be other issue - we will always have additional watcher for modelValue in every component when it is not needed.

Which is not optimal."

Reply:
If there are no cases where `stateful` can change dynamically, we can do this:
```
if (props.stateful) {
  watch(() => props.modelValue, (modelValue) => {
    if (modelValue !== undefined) {
      valueState.value = modelValue
    }
  }, { immediate: true })
}
```
otherwise we can update your solution a bit:
```
let unwatchModelValue: Function

const watchModelValue = () => {
  unwatchModelValue = watch(() => props.modelValue, (modelValue) => {
    if (modelValue !== undefined) {
      valueState.value = modelValue
    }
  }, { immediate: true }) // to handle if passed defaultValue
}

watch(() => props.stateful, (stateful) => {
  stateful ? watchModelValue() : unwatchModelValue?.()
}, { immediate: true })
```

Reply:
`Stateful` can be changed by the user any time.

`undefined` can be a valid `modelValue` in some cases.

Passed `defaultValue` is handled immediately here:
`  const valueState = ref(defaultValue === undefined ? props.modelValue : defaultValue) as Ref<T>`

I've moved `refs` to `functions` though, indeed slightly less verbose but less explicit as well.